tests/grpc/test_multi_thread.py

```diff
@@ -0,0 +1,148 @@
+import sys
+from typing import Optional
+
+import pinecone
+import threading
+from concurrent.futures import ThreadPoolExecutor, ProcessPoolExecutor
+import multiprocessing 
+import pytest
+from pinecone import Vector
+import pinecone 
+from ..utils.utils import index_fixture_factory
+from ..utils.remote_index import RemoteIndex, PodType
+import numpy as np
+import random 
+import time
+
+from ..unit.test_data_plane import get_test_data
+
+BATCH_SIZE = 100
+DIMENSION = 128
+TOP_K = 10
+
+test_mutli_thread = index_fixture_factory(
+    [
+         (RemoteIndex(pods=2, index_name=f'test-multi-thread-{PodType.P1}',
+                     dimension=DIMENSION, pod_type=PodType.P1), str(PodType.P1))
+    ]
+)
+
+def get_random_vectors(num, dim):
+    return [Vector(np.random.rand(dim).astype(np.float32).tolist()) for _ in range(num)]
+
+def make_upsert(index):
+    response = index.upsert(vectors=get_test_data(BATCH_SIZE, dimension=DIMENSION))
+    return response
+
+def make_query(index):
+    return index.query(values=np.random.rand(DIMENSION).astype(np.float32).tolist(),top_k=TOP_K)
+
+def check_upsert_results(results, num_expected):
+    total_vectors = 0
+    for res in results:
+        total_vectors += res.result().upserted_count
+    assert total_vectors == num_expected
+
+
+def test_multithreaded_upsert(test_mutli_thread):
+    index, _ = test_mutli_thread
+    num_threads = 10
+    with ThreadPoolExecutor(max_workers=num_threads) as executor:
+        results = []
+        for _ in range(num_threads):
+            results.append(executor.submit(make_upsert, index))
+        check_upsert_results(results, BATCH_SIZE * num_threads)
+
+
+def test_multithreaded_query(test_mutli_thread):
+    index, _ = test_mutli_thread
+    num_threads = 10
+    with ThreadPoolExecutor(max_workers=num_threads) as executor:
+        results = []
+        for _ in range(num_threads):
+            results.append(executor.submit(make_query, index))
+        for res in results:
+            assert len(res.result()) == TOP_K
+
+
+@pytest.mark.skip(reason="Multiprocessing using single Index object is currently not supported")
```

Comment:
instead of skipping, these should just do the separate index object per process thing

Reply:
not really a *native* way to support multiprocessing, but it's a decent temp. measure imo

Reply:
I think we should keep this test, as a "reminder" that this pattern is currently not supported.  
In the future we might be requested to support it.